pypeit/metadata.py

```diff
@@ -321,11 +321,22 @@ def _build(self, files, strict=True, usrdata=None):
                 data[meta_key].append(value)
             msgs.info('Added metadata for {0}'.format(os.path.split(ifile)[1]))
 
-#        # Additional bits and pieces
-#        self._add_bkg_pairs(data, 'empty')
+        # JFH Changed the below to now crash if some files have None in their MJD. This is the desired behavior
```

Comment:
"not" or "now"?

Reply:
PS - I worry a bit about things like this because these warning messages could easily fly by a user unnoticed.  We should think about if we're logging warnings in a way that a user can easily parse.

Reply:
Sorry it was supposed to be not. Anyway I think the code should crash when something is seriously wrong. I don't think a corrupt file should crash setup. I think if there is a corrupt file, it should be written to the .sorted file with some kind of warning or info. 

The issue I had was the following. I was running PypeIt while observing. The instrument controller would create an empty file to hold the current exposure and then update it with the data at readout. This meant that when I copied the data over I always had one empty file. This crashed setup because of the line above. The only work around was to constantly delete this one empty file. People running PypeIt while observing is something we want to aspire for, so I think having setup constantly crash if there is one file that is either empty or corrupt is a bad idea.

Reply:
So this try except block doesn't really have anything to do with the success of the `time.Time` call, right?  Shouldn't this check get put further up in the function?  And should we remove the incomplete/corrupt files from the internal metadata table?

Reply:
My suggestion would be to never crash in pypeit_setup. There is really no reason to crash there, and any crash was just bad coding. We are classifying files and assigning meta data. If pypeit_setup comes to something it cannot classify, then it should simply put that in a separate setup "Unknown" or something and leave the file in the .sorted list with a bunch of None values for the meta data.   This is enough warning for the user who can then figure out what went wrong. 

The try: except: block that I added was a stopgap to prevent the crash. I'm not sure why that line is there, it is just validating that the times are okay but not returning anything. It is exactly this "validation" which is in my opinion the flawed approach, because the code crashes when it fails to validate which is (in my opinion) not the desired behavior. 

Right now I'd like to leave this try:except in there to prevent crashes, but I'd be much happier if we made an Issue on not having pypeit_setup ever crash and instead just putting unclassifiable files in their own bin with None meta data.